File: app/buildOLD/tmp/kapt3/stubs/debug/com/chapelotas/app/data/DataModule.java
package com.chapelotas.app.data;

/**
 * Módulo de Hilt que provee las implementaciones de los repositorios
 */
@dagger.Module()
@kotlin.Metadata(mv = {1, 9, 0}, k = 1, xi = 48, d1 = {"\u0000<\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\b\'\u0018\u00002\u00020\u0001B\u0005\u00a2\u0006\u0002\u0010\u0002J\u0010\u0010\u0003\u001a\u00020\u00042\u0006\u0010\u0005\u001a\u00020\u0006H\'J\u0010\u0010\u0007\u001a\u00020\b2\u0006\u0010\t\u001a\u00020\nH\'J\u0010\u0010\u000b\u001a\u00020\f2\u0006\u0010\r\u001a\u00020\u000eH\'J\u0010\u0010\u000f\u001a\u00020\u00102\u0006\u0010\u0011\u001a\u00020\u0012H\'\u00a8\u0006\u0013"}, d2 = {"Lcom/chapelotas/app/data/DataModule;", "", "()V", "bindAIRepository", "Lcom/chapelotas/app/domain/repositories/AIRepository;", "aiRepositoryImpl", "Lcom/chapelotas/app/data/ai/AIRepositoryImpl;", "bindCalendarRepository", "Lcom/chapelotas/app/domain/repositories/CalendarRepository;", "calendarRepositoryImpl", "Lcom/chapelotas/app/data/calendar/CalendarRepositoryImpl;", "bindNotificationRepository", "Lcom/chapelotas/app/domain/repositories/NotificationRepository;", "notificationRepositoryImpl", "Lcom/chapelotas/app/data/notifications/NotificationRepositoryImpl;", "bindPreferencesRepository", "Lcom/chapelotas/app/domain/repositories/PreferencesRepository;", "preferencesRepositoryImpl", "Lcom/chapelotas/app/data/PreferencesRepositoryImpl;", "app_debug"})
@dagger.hilt.InstallIn(value = {dagger.hilt.components.SingletonComponent.class})
public abstract class DataModule {
    
    public DataModule() {
        super();
    }
    
    @dagger.Binds()
    @javax.inject.Singleton()
    @org.jetbrains.annotations.NotNull()
    public abstract com.chapelotas.app.domain.repositories.CalendarRepository bindCalendarRepository(@org.jetbrains.annotations.NotNull()
    com.chapelotas.app.data.calendar.CalendarRepositoryImpl calendarRepositoryImpl);
    
    @dagger.Binds()
    @javax.inject.Singleton()
    @org.jetbrains.annotations.NotNull()
    public abstract com.chapelotas.app.domain.repositories.NotificationRepository bindNotificationRepository(@org.jetbrains.annotations.NotNull()
    com.chapelotas.app.data.notifications.NotificationRepositoryImpl notificationRepositoryImpl);
    
    @dagger.Binds()
    @javax.inject.Singleton()
    @org.jetbrains.annotations.NotNull()
    public abstract com.chapelotas.app.domain.repositories.AIRepository bindAIRepository(@org.jetbrains.annotations.NotNull()
    com.chapelotas.app.data.ai.AIRepositoryImpl aiRepositoryImpl);
    
    @dagger.Binds()
    @javax.inject.Singleton()
    @org.jetbrains.annotations.NotNull()
    public abstract com.chapelotas.app.domain.repositories.PreferencesRepository bindPreferencesRepository(@org.jetbrains.annotations.NotNull()
    com.chapelotas.app.data.PreferencesRepositoryImpl preferencesRepositoryImpl);
}

File: app/src/main/java/com/chapelotas/app/data/DataModule.kt
package com.chapelotas.app.data

import android.content.Context
import com.chapelotas.app.data.ai.AIRepositoryImpl
import com.chapelotas.app.data.calendar.CalendarRepositoryImpl
import com.chapelotas.app.data.database.ChapelotasDatabase
import com.chapelotas.app.data.database.daos.*
import com.chapelotas.app.data.notifications.NotificationRepositoryImpl
import com.chapelotas.app.domain.events.ChapelotasEventBus
import com.chapelotas.app.domain.repositories.AIRepository
import com.chapelotas.app.domain.repositories.CalendarRepository
import com.chapelotas.app.domain.repositories.NotificationRepository
import com.chapelotas.app.domain.repositories.PreferencesRepository
import com.chapelotas.app.domain.usecases.*
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import dagger.Binds
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.SupervisorJob
import javax.inject.Singleton

/**
 * Módulo de Hilt - VERSIÓN FINAL con Room y todos los servicios
 */
@Module
@InstallIn(SingletonComponent::class)
abstract class DataModule {

    @Binds
    @Singleton
    abstract fun bindCalendarRepository(
        calendarRepositoryImpl: CalendarRepositoryImpl
    ): CalendarRepository

    @Binds
    @Singleton
    abstract fun bindNotificationRepository(
        notificationRepositoryImpl: NotificationRepositoryImpl
    ): NotificationRepository

    @Binds
    @Singleton
    abstract fun bindAIRepository(
        aiRepositoryImpl: AIRepositoryImpl
    ): AIRepository

    @Binds
    @Singleton
    abstract fun bindPreferencesRepository(
        preferencesRepositoryImpl: PreferencesRepositoryImpl
    ): PreferencesRepository

    companion object {

        // ===== ROOM DATABASE =====
        @Provides
        @Singleton
        fun provideDatabase(
            @ApplicationContext context: Context
        ): ChapelotasDatabase {
            return ChapelotasDatabase.getDatabase(
                context,
                CoroutineScope(SupervisorJob() + Dispatchers.IO)
            )
        }

        @Provides
        @Singleton
        fun provideDayPlanDao(database: ChapelotasDatabase): DayPlanDao {
            return database.dayPlanDao()
        }

        @Provides
        @Singleton
        fun provideEventPlanDao(database: ChapelotasDatabase): EventPlanDao {
            return database.eventPlanDao()
        }

        @Provides
        @Singleton
        fun provideNotificationDao(database: ChapelotasDatabase): NotificationDao {
            return database.notificationDao()
        }

        @Provides
        @Singleton
        fun provideConflictDao(database: ChapelotasDatabase): ConflictDao {
            return database.conflictDao()
        }

        @Provides
        @Singleton
        fun provideNotificationLogDao(database: ChapelotasDatabase): NotificationLogDao {
            return database.notificationLogDao()
        }

        // ===== CORE SERVICES =====
        @Provides
        @Singleton
        fun provideGson(): Gson {
            return GsonBuilder()
                .create()
        }

        @Provides
        @Singleton
        fun provideApplicationScope(): CoroutineScope {
            return CoroutineScope(SupervisorJob() + Dispatchers.Default)
        }

        @Provides
        @Singleton
        fun provideEventBus(): ChapelotasEventBus {
            return ChapelotasEventBus()
        }

        // ===== USE CASES =====
        @Provides
        @Singleton
        fun provideMasterPlanController(
            aiRepository: AIRepository,
            gson: Gson
        ): MasterPlanController {
            return MasterPlanController(aiRepository, gson)
        }

        @Provides
        @Singleton
        fun provideUnifiedMonkeyService(
            @ApplicationContext context: Context,
            database: ChapelotasDatabase,
            eventBus: ChapelotasEventBus,
            notificationRepository: NotificationRepository,
            masterPlanController: MasterPlanController,
            scope: CoroutineScope
        ): UnifiedMonkeyService {
            return UnifiedMonkeyService(
                context = context,
                database = database,
                eventBus = eventBus,
                notificationRepository = notificationRepository,
                masterPlanController = masterPlanController,
                scope = scope
            )
        }

        @Provides
        @Singleton
        fun provideNotificationActionHandler(
            database: ChapelotasDatabase,
            eventBus: ChapelotasEventBus,
            notificationManagerCompat: androidx.core.app.NotificationManagerCompat,
            scope: CoroutineScope
        ): NotificationActionHandler {
            return NotificationActionHandler(
                database = database,
                eventBus = eventBus,
                notificationManager = notificationManagerCompat,
                scope = scope
            )
        }

        @Provides
        @Singleton
        fun provideNotificationManagerCompat(
            @ApplicationContext context: Context
        ): androidx.core.app.NotificationManagerCompat {
            return androidx.core.app.NotificationManagerCompat.from(context)
        }

        // ===== LEGACY - Para compatibilidad temporal =====
        @Provides
        @Singleton
        fun provideMonkeyCheckerService(
            @ApplicationContext context: Context,
            masterPlanController: MasterPlanController,
            notificationRepository: NotificationRepository,
            gson: Gson
        ): MonkeyCheckerService {
            return MonkeyCheckerService(
                context = context,
                masterPlanController = masterPlanController,
                notificationRepository = notificationRepository,
                gson = gson
            )
        }
    }
}